frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx

```diff
@@ -0,0 +1,168 @@
+import { TableBase } from '~/components/table';
+import { usePipelineVersionLoadMore } from '~/concepts/pipelines/content/tables/usePipelineLoadMore';
+import { PipelineVersionKF } from '~/concepts/pipelines/kfTypes';
+import { pipelineVersionSelectorColumns } from '~/concepts/pipelines/content/pipelineSelector/columns';
+import usePipelineVersionsTable from '~/concepts/pipelines/content/tables/pipelineVersion/usePipelineVersionsTable';
+import PipelineViewMoreFooterRow from '~/concepts/pipelines/content/tables/PipelineViewMoreFooterRow';
+import {
+  getSelectorTableColumnSort,
+  useSelectorSearch,
+} from '~/concepts/pipelines/content/pipelineSelector/utils';
+import EmptyTableView from '~/concepts/pipelines/content/tables/EmptyTableView';
+
+type PipelineVersionSelectorProps = {
+  pipelineId?: string;
+  selection?: string;
+  onSelect: (version: PipelineVersionKF) => void;
+};
+
+const PipelineVersionSelector: React.FC<PipelineVersionSelectorProps> = ({
+  pipelineId,
+  selection,
+  onSelect,
+}) => {
+  const [isOpen, setOpen] = React.useState(false);
+
+  const [
+    [{ items: initialData, totalSize: fetchedSize, nextPageToken: initialPageToken }, loaded],
+    { initialLoaded, ...tableProps },
+  ] = usePipelineVersionsTable(pipelineId)();
+
+  const toggleRef = React.useRef(null);
+  const menuRef = React.useRef(null);
+
+  const { sortDirection, sortField, setFilter, filter } = tableProps;
+  const { data: versions, onLoadMore } = usePipelineVersionLoadMore({
+    pipelineId,
+    initialData,
+    initialPageToken,
+    sortDirection,
+    sortField,
+    filter,
+    loaded,
+  });
+
+  const { totalSize, ...searchProps } = useSelectorSearch({ setFilter, fetchedSize, loaded });
+  const { onClear } = searchProps;
+
+  const menu = (
+    <Menu data-id="pipeline-version-selector-menu" ref={menuRef} isScrollable>
+      <MenuContent>
+        <MenuSearch>
+          <MenuSearchInput>
+            <SearchInput {...searchProps} aria-label="Filter pipeline versions" />
+          </MenuSearchInput>
+          <HelperText>
+            <HelperTextItem variant="indeterminate">{`Type a name to search your ${totalSize} versions.`}</HelperTextItem>
+          </HelperText>
+        </MenuSearch>
+        <MenuList>
+          <div role="menuitem">
+            <TableBase
+              {...tableProps}
+              itemCount={fetchedSize}
+              loading={!loaded}
+              data-id="pipeline-version-selector-table-list"
+              emptyTableView={
+                <EmptyTableView
+                  hasIcon={false}
+                  onClearFilters={onClear}
+                  variant={EmptyStateVariant.xs}
+                />
+              }
+              borders={false}
+              variant={TableVariant.compact}
+              columns={pipelineVersionSelectorColumns}
+              data={versions}
+              rowRenderer={(row, index) => (
+                <PipelineSelectorTableRow
+                  key={index}
```

Comment:
Use `row.id` (or something else) instead of `index`.
```suggestion
                  key={row.id}
```